mplc/experiment.py

```diff
@@ -0,0 +1,138 @@
+        """Create scenarios from a config file passed as argument,
+           and populates scenarios list accordingly"""
+
+        config = utils.get_config_from_file(path_to_config_file)
+        scenario_params_list = utils.get_scenario_params_list(config["scenario_params_list"])
+
+        for scenario_params_idx, scenario_params in enumerate(scenario_params_list):
+
+            scenario_params_index_str = f"{scenario_params_idx + 1}/{len(scenario_params_list)}"
+            logger.info(f"Creating scenarios from config file")
+            logger.debug(f"Scenario {scenario_params_index_str}: {scenario_params}")
+
+            current_scenario = scenario_module.Scenario(
+                **scenario_params,
+                experiment_path=self.experiment_path,
+                scenario_id=scenario_params_idx+1,
+            )
+
+            self.add_scenario(current_scenario)
+
+    def validate_scenarios_list(self):
+        """Instantiate every scenario without running it to check if
+           every scenario is correctly specified. This prevents scenario initialization errors during the experiment"""
+
+        logger.debug("Starting validation of scenarios prior to running the experiment")
+
+        for scenario_idx, scenario in enumerate(self.scenarios_list):
+
+            scenario_index_str = f"{scenario_idx + 1}/{len(self.scenarios_list)}"
+            logger.debug(f"Scenarios validation: now validating scenario {scenario_index_str} "
+                         f"(instantiate partners, split data, compute batch sizes, apply data corruption req.)")
+
+            scenario.run(is_dry_run=True)
+
+        logger.debug("Scenarios validation: all scenario have been validated successfully")
+
+    def run_experiment(self):
+        """Run the experiment, starting by validating the scenarios first"""
+
+        # Preliminary steps
+        plt.close("all")  # Close open figures
+        utils.set_log_file(self.experiment_path)  # Move log files to experiment folder
+
+        # Validate scenarios prior to running the experiment
+        self.validate_scenarios_list()
+
+        # Loop over nb_repeats
+        for repeat_idx in range(self.nb_repeats):
+
+            repeat_index_str = f"{repeat_idx + 1}/{self.nb_repeats}"
+            logger.info(f"Now starting repeat {repeat_index_str}")
+
+            # Loop over scenarios in scenarios_list
+            for scenario_idx, scenario in enumerate(self.scenarios_list):
+
+                scenario_index_str = f"{scenario_idx + 1}/{len(self.scenarios_list)}"
+                logger.info(f"(Repeat {repeat_index_str}) Now running scenario {scenario_index_str}")
+
+                # Run the scenario
+                scenario.n_repeat = repeat_idx
+                scenario.run()
+
+                # Save scenario results
+                df_results = scenario.to_dataframe()
+                df_results["repeat_index"] = repeat_idx
+                df_results["scenario_index"] = scenario_idx
+
```

Comment:
We can maybe add somewhere an other .csv per scenario, with the result of the mpl ? from the history object. Otherwise, these will be discarded. 
It can also be a choice to consider these results as too detailed to be stored by an experiment.
I don't know

Reply:
I think that is a good idea

Reply:
@arthurPignet does the `History` object has a method to print it / write it to disk?

Reply:
yes, history.save_data() (which is a refacto for the old ml.save_data())
But it saves the dict history.history in binary, I think we should change that to save the dataframe 
The method also save some plots.